src/compiler/crystal/codegen/call.cr

```diff
@@ -360,7 +360,11 @@ class Crystal::CodeGenVisitor
           end
           node.args.each_with_index do |node_arg, i|
             a_def_arg = a_def.args[i]
-            result = and(result, match_type_id(node_arg.type, a_def_arg.type, arg_type_ids[i]))
+            if autocast_literal?(node_arg)
+              # Matches, so nothing to do
+            else
+              result = and(result, match_type_id(node_arg.type, a_def_arg.type, arg_type_ids[i]))
+            end
```

Comment:
Why not simplify it?

```suggestion
            unless autocast_literal?(node_arg)
              result = and(result, match_type_id(node_arg.type, a_def_arg.type, arg_type_ids[i]))
            end
```

Reply:
Because I wanted to keep the comment?

Reply:
I thought you could add it above `unless` but ok.